src/test/java/com/synopsys/integration/detect/battery/BatteryTest.java

```diff
@@ -0,0 +1,319 @@
+        executables.add(BatteryExecutable.propertyOverrideExecutable(detectProperty, resourceCopyingExecutable));
+        return resourceCopyingExecutable;
+    }
+
+    public ResourceCopyingExecutableCreator executableSourceFileThatCopiesFiles(final String windowsName, final String linuxName, final String... resourceFiles) {
+        final ResourceCopyingExecutableCreator resourceCopyingExecutable = new ResourceCopyingExecutableCreator(prefixResources(resourceFiles));
+        executables.add(BatteryExecutable.sourceFileExecutable(windowsName, linuxName, resourceCopyingExecutable));
+        return resourceCopyingExecutable;
+    }
+
+    public void executable(final DetectProperty detectProperty, final String... responses) {
+        executables.add(BatteryExecutable.propertyOverrideExecutable(detectProperty, new StringTypingExecutableCreator(Arrays.asList(responses))));
+    }
+
+    public void git(final String origin, final String branch) {
+        sourceFileNamed(".git");
+        executable(DetectProperty.DETECT_GIT_PATH, origin, branch);
+    }
+
+    public void sourceFileNamed(final String filename) {
+        emptyFileNames.add(filename);
+    }
+
+    public void sourceFolderFromExpandedResource(final String filename) {
+        resourceZipNames.add(filename);
+    }
+
+    public void sourceFileFromResource(final String filename) {
+        resourceFileNames.add(filename);
+    }
+
+    public void expectBdioFile(final String named, final int size) {
+        bdioSizes.put(named, size);
+    }
+
+    public void expectBdioResources() {
+        shouldExpectBdioResources = true;
+    }
+
+    public void property(final DetectProperty property, final String value) {
+        property(property.getPropertyKey(), value);
+    }
+
+    public void property(final String property, final String value) {
+        additionalProperties.add("--" + property + "=" + value);
+    }
+
+    public void run() {
+        try {
+            checkEnvironment();
+            initializeDirectories();
+            createFiles();
+            final List<String> executableArguments = createExecutables();
+            runDetect(executableArguments);
+
+            assertBdio();
+        } catch (final ExecutableRunnerException | IOException | JSONException | TemplateException e) {
+            Assertions.assertNull(e, "An exception should not have been thrown!");
+        }
+    }
+
+    private void runDetect(final List<String> additionalArguments) throws IOException, ExecutableRunnerException {
+        final List<String> detectArguments = new ArrayList<>();
+        detectArguments.addAll(Arrays.asList("--detect.tools=DETECTOR", "--blackduck.offline.mode=true", "--detect.output.path=" + outputDirectory, "--detect.bdio.output.path=" + bdioDirectory, "--detect.cleanup=false"));
+
+        detectArguments.add("--logging.level.detect=DEBUG");
+        detectArguments.add("--detect.source.path=" + sourceDirectory.getCanonicalPath());
```

Comment:
Not in love with the properties being hardcoded. Can we reference DetectProperty directly?